lib/testutils/e2e_describe.go

```diff
@@ -0,0 +1,56 @@
+package testutils
+
+import (
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/etcd"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s"
+
+	. "github.com/onsi/ginkgo"
+)
+
+const (
+	ClientEtcdV2 = 1 << iota
+	ClientK8s
+)
+
+// E2eDescribe is a replacement for ginkgo.Describe which invoke Describe
+// multiple times for one or more different backend clients - passing in the
+// Calico API configuration as a parameter to the test function.  This allows
+// easy construction of end-to-end tests covering multiple different backends.
+//
+// The *clients* parameter is a bit-wise OR of the required client/backend
+// types that will be tested.
+func E2eDescribe(description string, clients int, body func(config api.CalicoAPIConfig)) bool {
```

Comment:
Renamed, and tweaked the wording to talk about datastore drivers rather than backends.